runtime/convertValues_test.go

```diff
@@ -2961,6 +3374,339 @@ func TestRuntimeStringValueImport(t *testing.T) {
 	})
 }
 
+func TestTypeValueImport(t *testing.T) {
+
+	t.Parallel()
+
+	t.Run("Type<Int>", func(t *testing.T) {
+
+		t.Parallel()
+
+		typeValue := cadence.NewTypeValue(cadence.IntType{})
+
+		script := `
+            pub fun main(s: Type) {
+                log(s.identifier)
+            }
+        `
+
+		encodedArg, err := json.Encode(typeValue)
+		require.NoError(t, err)
+
+		rt := NewInterpreterRuntime()
+
+		runtimeInterface := &testRuntimeInterface{
+			decodeArgument: func(b []byte, t cadence.Type) (value cadence.Value, err error) {
+				return json.Decode(b)
+			},
+			log: func(s string) {
+				assert.Equal(t, s, "\"Int\"")
+			},
+		}
+
+		_, err = rt.ExecuteScript(
+			Script{
+				Source:    []byte(script),
+				Arguments: [][]byte{encodedArg},
+			},
+			Context{
+				Interface: runtimeInterface,
+				Location:  TestLocation,
+			},
+		)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("missing", func(t *testing.T) {
+
+		t.Parallel()
+
+		typeValue := cadence.NewTypeValue(&cadence.StructType{
+			QualifiedIdentifier: "S",
+			Location:            TestLocation,
+			Fields:              []cadence.Field{},
+			Initializers:        [][]cadence.Parameter{},
+		})
+
+		script := `
+            pub fun main(s: Type) {
+                log(s.identifier)
+            }
+        `
+
+		encodedArg, err := json.Encode(typeValue)
+		require.NoError(t, err)
+
+		rt := NewInterpreterRuntime()
+
+		runtimeInterface := &testRuntimeInterface{
+			decodeArgument: func(b []byte, t cadence.Type) (value cadence.Value, err error) {
+				return json.Decode(b)
+			},
+			log: func(s string) {
+			},
+		}
+
+		_, err = rt.ExecuteScript(
+			Script{
+				Source:    []byte(script),
+				Arguments: [][]byte{encodedArg},
+			},
+			Context{
+				Interface: runtimeInterface,
+				Location:  TestLocation,
+			},
+		)
+
+		require.Error(t, err)
+	})
+}
+
+func TestCapabilityValueImport(t *testing.T) {
+
+	t.Parallel()
+
+	t.Run("public Capability<&Int>", func(t *testing.T) {
+
+		t.Parallel()
+
+		capabilityValue := cadence.Capability{
+			BorrowType: cadence.ReferenceType{Type: cadence.IntType{}},
+			Address:    cadence.Address{0x1},
+			Path: cadence.Path{
+				Domain:     common.PathDomainPublic.Identifier(),
+				Identifier: "foo",
+			},
+		}
+
+		script := `
+            pub fun main(s: Capability<&Int>) {
+                log(s)
+            }
+        `
+
+		encodedArg, err := json.Encode(capabilityValue)
+		require.NoError(t, err)
+
+		rt := NewInterpreterRuntime()
+
+		runtimeInterface := &testRuntimeInterface{
+			decodeArgument: func(b []byte, t cadence.Type) (value cadence.Value, err error) {
+				return json.Decode(b)
+			},
+			log: func(s string) {
+				assert.Equal(t, s, "Capability<&Int>(address: 0x100000000000000, path: /public/foo)")
```

Comment:
Similar to above, might want to add another assertion to make sure this gets executed